src/utils/fetch.spec.ts

```diff
@@ -93,15 +93,364 @@ describe('utils fetch', () => {
+                    if (headerName === 'content-type') {
+                        return 'application/json';
+                    }
+                    return undefined;
+                },
+            },
+            status: undefined,
+            text: () => {
+                return Promise.resolve('{}');
+            },
+        };
+
+        const promise = convertFetchSuccess(
+            'url',
+            'body',
+            0,
+            // @ts-ignore
+            result,
+        );
+
+        await expect(promise).rejects.toMatchInlineSnapshot(`
+                    Object {
+                      "headers": Object {
+                        "get": [Function],
+                      },
+                      "isNetworkError": true,
+                      "networkErrorType": "no-status",
+                      "requestId": undefined,
+                      "response": Object {},
+                      "responseType": "json",
+                      "size": 2,
+                      "status": undefined,
+                      "url": "url",
+                    }
+                `);
+    });
+
+    it('rejects with proxy error', async () => {
+        // @ts-ignore
+        const result = {
+            headers: {
+                get(headerName: string) {
+                    if (headerName === 'content-type') {
+                        return 'application/json';
+                    }
+                    return undefined;
+                },
+            },
+            status: 407,
+            text: () => {
+                return Promise.resolve('{}');
+            },
+        };
+
+        const promise = convertFetchSuccess(
+            'url',
+            'body',
+            0,
+            // @ts-ignore
+            result,
+        );
+
+        await expect(promise).rejects.toMatchInlineSnapshot(`
+                    Object {
+                      "headers": Object {
+                        "get": [Function],
+                      },
+                      "isNetworkError": true,
+                      "networkErrorType": "proxy-error",
+                      "requestId": undefined,
+                      "response": Object {},
+                      "responseType": "json",
+                      "size": 2,
+                      "status": 407,
+                      "url": "url",
+                    }
+                `);
+    });
+
+    it('rejects with akamai error', async () => {
+        // @ts-ignore
+        const result = {
+            headers: {
+                get(headerName: string) {
+                    if (headerName === 'content-type') {
+                        return 'html';
+                    }
+                    return undefined;
+                },
+            },
+            status: 503,
+            text: () => {
+                return Promise.resolve(
+                    '<html>Akami Error Reference&#32;83486234237498239423949238</html>',

```

Comment:
nitpick - typo in `Akami`, not that it matters, so feel free to close it/fix later